app/core/database_service.py

```
@@ -0,0 +1,560 @@
+        except Exception as e:
+            raise DatabaseError(
+                message=f"Failed to delete {model.__name__}: {str(e)}",
+                operation="delete_record",
+                model=model.__name__,
+                context={"id": str(id_value)}
+            ) from e
+    
+    async def execute_raw_query(
+        self,
+        query: str,
+        params: Dict[str, Any] = None,
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)